src/test/java/org/terasology/launcher/repositories/JenkinsRepositoryAdapterTest.java

```diff
@@ -0,0 +1,205 @@
+                "}";
+    }
+
+    static String incompleteArtifactsPayload() {
+        return "{\n" +
+                "  \"builds\": [\n" +
+                "    {\n" +
+                "      \"artifacts\": [\n" +
+                "        {\n" +
+                "          \"fileName\": \"versionInfo.properties\",\n" +
+                "          \"relativePath\": \"distros/omega/versionInfo.properties\"\n" +
+                "        }\n" +
+                "      ],\n" +
+                "      \"number\": 1,\n" +
+                "      \"result\": \"SUCCESS\",\n" +
+                "      \"timestamp\": 1604285977306,\n" +
+                "      \"url\": \"http://jenkins.terasology.io/teraorg/job/Nanoware/job/Omega/job/develop/1/\"\n" +
+                "    }\n" +
+                "  ]\n" +
+                "}";
+    }
+
+    static List<String> incompletePayloads() {
+        return List.of(
+                "{}",
+                "{ \"builds\": [] }",
+                nullArtifactsPayload(),
+                emptyArtifactsPayload(),
+                incompleteArtifactsPayload()
+        );
+    }
+
+    static Stream<Arguments> incompleteResults() {
+        return incompleteResults.stream().map(Arguments::of);
+    }
+
+    @Test
+    @DisplayName("handle null Jenkins response gracefully")
+    void shouldHandleNullJenkinsResponseGracefully() {
+        final JenkinsClient nullClient = new StubJenkinsClient(url -> null, url -> null);
+        final JenkinsRepositoryAdapter adapter = new JenkinsRepositoryAdapter(Profile.OMEGA, Build.STABLE, nullClient);
+        assertTrue(adapter.fetchReleases().isEmpty());
+    }
+
+    @Test
+    @DisplayName("skip builds without version info")
+    void shouldSkipBuildsWithoutVersionInfo() {
+        Properties emptyVersionInfo = new Properties();
+
+        final JenkinsClient stubClient = new StubJenkinsClient(url -> validResult, url -> emptyVersionInfo);
+
+        final JenkinsRepositoryAdapter adapter = new JenkinsRepositoryAdapter(Profile.OMEGA, Build.STABLE, stubClient);
+
+        assertTrue(adapter.fetchReleases().isEmpty());
+    }
+
+    @Test
+    @DisplayName("process valid response correctly")
+    void shouldProcessValidResponseCorrectly() {
+        String expectedVersion = "alpha 42 (preview) - 20210130";
+
+        Properties versionInfo = new Properties();
+        versionInfo.setProperty("displayVersion", expectedVersion);
+
+        final JenkinsClient stubClient = new StubJenkinsClient(url -> validResult, url -> versionInfo);
+
+        final GameIdentifier id = new GameIdentifier(expectedVersion, Build.STABLE, Profile.OMEGA);
+        final GameRelease expected = new GameRelease(id, expectedArtifactUrl, new ArrayList<>(), new Date(1604285977306L));
+
+        final JenkinsRepositoryAdapter adapter = new JenkinsRepositoryAdapter(Profile.OMEGA, Build.STABLE, stubClient);
+
+        assertAll(
+                () -> assertEquals(1, adapter.fetchReleases().size()),
```

Comment:
Please extract this from the `assertAll` otherwise the following asserts may break the test in a not graceful way